transport/ws/src/main/java/org/kaazing/gateway/transport/ws/extension/WebSocketExtensionFactorySpi.java

```diff
@@ -56,4 +56,33 @@
     public abstract WebSocketExtensionSpi negotiate(ExtensionHeader requestedExtension, WsResourceAddress address)
             throws ProtocolException;
 
+    /**
+     * This method allows extensions to specify the order that they would like to be placed on the Filter Chain.
+     *
+     * @return A {@link WebSocketExtensionFactorySpi.ExtensionOrderCategory} that specifies the extension type and where it
+     * should be ordered
+     */
+    public ExtensionOrderCategory orderCategory() {
+        return ExtensionOrderCategory.OTHER;
+    }
+
+    /**
+     * An ordered enum that describes the category that an extension requests to be ordered in.  The enum is ordered by closest
+     * to the network
+     */
+    enum ExtensionOrderCategory{
+        /**
+         * Network, meaning something that transforms the network bytes
+         */
+        NETWORK,
+        /**
+         * Emulation, meaning emulating a frame type that should not be exposed to the application
+         */
+        EMULATION,
```

Comment:
Consider renaming to "FRAMING" since this category concerns extension which change the websocket frame flow.